build_wheels.py

```diff
@@ -172,28 +177,239 @@ def assemble_requirements(idf_branches: List[str], idf_constraints: List[str]) -
+        yaml_list = yaml.load(f, yaml.Loader)
+
+    requirements_set: set[Requirement] = set()
+
+    if not yaml_list:
+        return requirements_set
+
+    for package in yaml_list:
+        requirement_txt = f"{package['package_name']}"
+
+        if 'version' in package:
+            if not isinstance(package['version'], list):
+                requirement_txt += _change_specifier_logic(package['version']) if exclude else package['version']
+
+            if isinstance(package['version'], list):
+                version_list = (
+                    [f'{_change_specifier_logic(ver)}' if exclude else f'{ver}' for ver in package['version']]
+                    )
+
+                requirement_txt += ','.join(version_list)
+
+
+        if 'platform' in package and 'version' not in package:
+            if not isinstance(package['platform'], list):
+                requirement_txt += (
+                    f"; sys_platform != '{package['platform']}'" if exclude
+                    else f"; sys_platform == '{package['platform']}'"
+                    )
+
+            if isinstance(package['platform'], list):
+                platform_list = (
+                    [f"sys_platform != '{plf}'" if exclude
+                     else f"sys_platform == '{plf}'" for plf in package['platform']]
+                    )
+
+                requirement_txt += '; ' + ' or '.join(platform_list)
+
+
+        if 'platform' in package and 'version' in package:
+            requirement_old = f"{package['package_name']}"
+
+            if not isinstance(package['platform'], list):
+                requirement_txt += f"; sys_platform == '{package['platform']}'"
+                if exclude:
+                    requirement_old += f"; sys_platform != '{package['platform']}'"
+                    requirements_set.add(Requirement(requirement_old))
+
+            if isinstance(package['platform'], list):
+                platform_list = [f"sys_platform == '{plf}'" for plf in package['platform']]
+                requirement_txt += '; ' + ' or '.join(platform_list)
+
+                if exclude:
+                    platform_list_old =  [f"sys_platform != '{plf}'" for plf in package['platform']]
+                    requirement_old += '; ' + ' or '.join(platform_list_old)
+                    requirements_set.add(Requirement(requirement_old))
+
+
+        requirements_set.add(Requirement(requirement_txt))
+    return requirements_set
+
+
+def exclude_from_requirements(assembled_requirements:set, exclude_list: set, print_requirements: bool = True) -> set:
+    """Exclude packages defined in exclude_list from assembled requirements
+        - print_requirements = true will print the changes
+    """
+    new_assembled_requirements = set()
+    not_in_exclude = []
+    if print_requirements:
+        print(Fore.BLUE + '---------- REQUIREMENTS ----------')
+
+    for requirement in assembled_requirements:
+        printed = False
+        for req in exclude_list:
```

Comment:
This function is very hard to read because of the nesting level ... it might be worth rethinking this design (split the logic into smaller blocks).

In addition, using `requirement` for the outer loop and a look-alike `req` for the inner loop is extra confusing ... a longer more descriptive variable name would be better here.